### ISSUE TITLE
CA1820 does not suggest code fix for string.Equals

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/roslyn-analyzers#4537]
### Analyzer

**Diagnostic ID**: [CA1820](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1820): `Test for empty strings using string lengths`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

A CA1820 rule violation does not suggest a code fix when using the `string.Equals` method.

### Steps To Reproduce

```csharp
s.Equals(string.Empty)
```

### Expected behavior

Code fix is offered:

```csharp
s.Length == 0
```

### Actual behavior

No code fix is offered

**Comment by Evangelink** [comment]
Currently the code-fix is only offered for `==` and `!=`, we could improve the code-fix to also handle `.Equals`.

**Comment by xtqqczze** [comment]
We would use a different code fix: `s.Length == 0` as we can assume `s` is not null.

**Comment by mavasani** [comment]
Please target https://github.com/dotnet/roslyn-analyzers/tree/release/6.0.1xx-preview1 branch as this would be a new feature.

**Comment by xtqqczze** [comment]
@mavasani Just to be completely clear, `s.Equals(string.Empty)` is already identified as a `CA1820` violation but no code fix is offered.

**Comment by mavasani** [comment]
@xtqqczze Ah, thanks. So, this is just a bug fix to an existing code fix? If so, targeting master should also be fine.

**Comment by xtqqczze** [comment]
@mavasani I think it is a bad experience to report the violation without offering a fix, especially considering this fix is unambiguous. I am not familiar with repository policy, so would you consider this a bug fix, or not ?

**Comment by mavasani** [comment]
@xtqqczze Yes, it would be fine to be considered as a bug fix that targets the master branch. Please feel free to work on it and submit a PR.

**Comment by xtqqczze** [comment]
I have not worked on code analyzer development before, so I cannot commit to submitting a PR, at least in the short term.

**Comment by Evangelink** [comment]
It might not be the simplest to start with but if you want to give it a try, I am happy to support you.

**Comment by hrkeni** [comment]
I would like to take a crack at this

**Comment by xtqqczze** [comment]
@hrkeni It would be great if you could. I am happy to help with tests.
